plugins/spectrum_arbscor.py

```diff
@@ -0,0 +1,228 @@
+        37.48.
+        :param rep:
+        :return:
+        """
+        order = []
+        # 3x3 base
+        for y_base in range(3):
+            for x_base in range(3):
+                # Every 3rd pixel (within the repetition)
+                for j in range(y_base, rep[1], 3):  # Y slow
+                    for i in range(x_base, rep[0], 3):  # X fast
+                        order.append((i, j))
+
+        return order
+
+    def _get_scan_order(self, rep: Tuple[int, int]) -> List[Tuple[int, int]]:
+        """
+        Computes the arbitrary scan order
+        :param rep: x,y number of pixels to scan
+        :return: list of positions (x, y) as indices to scan, in the scan order
+        """
+        #return self._checkboard_scan_order(rep)
+        return self._33_scan_order(rep)
+
+    def _getSpotPositions(self):
+        """
+        Compute the positions of the e-beam for each point in the ROI
+        return (numpy ndarray of floats of shape (Y,X,2)): each value is for a
+          given Y,X in the rep grid -> 2 floats corresponding to the
+          translation X,Y. Note that the dimension order is different between
+          index and content, because X should be scanned first, so it's last
+          dimension in the index.
+        """
+        rep = tuple(self.repetition.value)
+        roi = self.roi.value
+        width = (roi[2] - roi[0], roi[3] - roi[1])
+
+        # Take into account the "border" around each pixel
+        pxs = (width[0] / rep[0], width[1] / rep[1])
+        lim = (roi[0] + pxs[0] / 2, roi[1] + pxs[1] / 2)
+
+        shape = self._emitter.shape
+        # convert into SEM translation coordinates: distance in px from center
+        # (situated at 0.5, 0.5), can be floats
+        pos00 = (shape[0] * (lim[0] - 0.5), shape[1] * (lim[1] - 0.5))
+        logging.debug("Generating points from %s with pxs %s, from rep %s and roi %s",
+                      pos00, pxs, rep, roi)
+
+        pos = numpy.empty((rep[1], rep[0], 2), dtype=float)
+        order = self._get_scan_order(rep)
+        logging.debug("Arbitrary order: %s", order)
+        self._px_order = order
+
+        raw_order = numpy.ndindex(*rep[::-1])
+        for raw_idx, px_idx in zip(raw_order, order):
+            beam_pos = (pos00[0] + px_idx[0] * pxs[0],
+                        pos00[1] + px_idx[1] * pxs[1],)
+            pos[raw_idx] = beam_pos
+
+        logging.debug("Will acquire at beam pos: %s", pos)
+        return pos
+
+    def _assembleLiveData(self, n, raw_data, px_idx, rep, pol_idx):
+        """
+        :param px_idx: y, x position
+        :param rep: x, y number of points in teh scan
+        """
+        # px_idx
+        px_idx_flat = px_idx[0] * rep[0] + px_idx[1]
+        act_px_idx = self._px_order[px_idx_flat][::-1]
+        logging.debug("Converted back idx %s to %s", px_idx, act_px_idx)
+        return super()._assembleLiveData(n, raw_data, act_px_idx, rep, pol_idx)
+
+
+
+
+class SpectrumArbScOrPlugin(Plugin):
```

Comment:
class name looks a bit silly, perhaps just change it to SpectrumArbitraryScanOrderPlugin?

Reply:
I was very proud of an abbreviated name that can still be pronounce, and you call it silly! :wink: 
Ah... well I'll put the full name. It's actually fine too.

Reply:
Haha yeah well with camelcase notation it looked a bit silly, sorry for that.